pilot/pkg/serviceregistry/kube/controller/ambient/policies.go

```diff
@@ -59,6 +61,15 @@ func WaypointPolicyStatusCollection(
 				reason := "unknown"
 				bound := false
 				switch target.GetKind() {
+				case gvk.GatewayClass_v1.Kind:
+					fetchedGatewayClasses := krt.Fetch(ctx, gatewayClasses, krt.FilterKey(target.GetName()))
+					if len(fetchedGatewayClasses) == 1 {
+						bound = true
+						reason = model.WaypointPolicyReasonAccepted
+						message = fmt.Sprintf("bound to %s", fetchedGatewayClasses[0].GetName())
+					} else {
+						reason = model.WaypointPolicyReasonTargetNotFound
```

Comment:
nit: can we add a log line here so we have some evidence that there are duplicate waypoints in the collection? It shouldn't be possible in single cluster, but would be evidence of a bug once we get to multicluster